web-admin/src/routes/[organization]/[project]/-/redirect/+page.svelte

```diff
@@ -1,13 +1,33 @@
+<!-- 
+  This page is used to redirect to either the project's first dashboard, or to the project's status page.
+ -->
 <script lang="ts">
   import { goto } from "$app/navigation";
   import { page } from "$app/stores";
   import { useDashboardNames } from "@rilldata/web-common/features/dashboards/selectors";
   import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+  import { createAdminServiceGetProject } from "../../../../../client";
 
-  // Currently, we don't have an `/[organization]` page. Instead, `/[organization]` routes to the organization's first project, and ideally that project's first dashboard.
-  // This page is used to redirect to either the project's first dashboard, or to the project's status page.
-  $: dashboardsQuery = useDashboardNames($runtime.instanceId);
-  $: if ($dashboardsQuery.isSuccess) {
+  $: proj = createAdminServiceGetProject(
+    $page.params.organization,
+    $page.params.project,
+    {
+      query: {
+        placeholderData: undefined,
+      },
+    }
+  );
+
+  // Avoid a race condition: ensure the runtime store has been updated with the new project's runtime instance ID
+  $: isRuntimeStoreReady =
+    $proj?.data &&
+    $proj.data.prodDeployment.runtimeInstanceId === $runtime.instanceId;
+
+  let dashboardsQuery;
+  $: if (isRuntimeStoreReady) {
+    dashboardsQuery = useDashboardNames($runtime.instanceId);
```

Comment:
Why not use `$proj.data.prodDeployment.runtimeInstanceId` here instead of dealing with race condition?

Reply:
Because `useDashboardNames`'s `httpClient` needs `$runtime.host` and `$runtime.jwt`. If we use `$proj.data.prodDeployment.runtimeInstanceId`, then the `httpClient` might reference `$runtime` before the new `host` and `jwt` are set.

Reply:
Just tried to clarify the comment a bit